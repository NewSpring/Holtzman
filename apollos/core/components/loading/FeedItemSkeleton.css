.load-item {
  background: #dddddd;
}

.fake-text,
.fake-text-small {
  /* force hardware acceleration */
  -webkit-transform: translate3d(0, 0, 0);

  background: linear-gradient(270deg, #dddddd, #bcbcbc, #dddddd);
  -webkit-animation: ColorAnimation 5s cubic-bezier(0.66, 0.21, 0.49, 0.88) infinite;
  animation: ColorAnimation 5s cubic-bezier(0.66, 0.21, 0.49, 0.88) infinite;
  -ms-animation: ColorAnimation 5s cubic-bezier(0.66, 0.21, 0.49, 0.88) infinite;
  background-size: 200%;

}

@keyframes ColorAnimation {
    0%{background-position:100%}

    100%{background-position:-100%}
}

@-webkit-keyframes ColorAnimation {
    0%{background-position:100%}

    100%{background-position:-100%}
}

@-ms-keyframes ColorAnimation {
    0%{background-position:100%}

    100%{background-position:-100%}
}

.fake-text {
  height: 15px;
}

.fake-text-small {
  height: 8px;
}
.fake-text, .fake-text-small{
  border-radius: 2px;
}

:global(.imageloader > :first-child) {
  opacity: 0.5;
}

:global(.imageloader.loaded > :first-child) {
  /* force hardware acceleration */
  -webkit-transform: translate3d(0, 0, 0);

  opacity: 1;
  transition: opacity .5s ease-in-out;
 -moz-transition: opacity .5s ease-in-out;
 -webkit-transition: opacity .5s ease-in-out;
}
