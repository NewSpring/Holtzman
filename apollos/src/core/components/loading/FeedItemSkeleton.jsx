import { Component } from "react"

import Styles from "./FeedItemSkeleton.css"

export default class FeedItemSkeleton extends Component {

  backgroundStyles = () => {
    let classes = [
      "rounded-top",
      "ratio--square",
      "background--light-tertiary",
      Styles["load-item"]
    ];

    return classes.join(" ")
  }

  titleStyles = () => {
    let classes = [
      "one-whole",
      "push-half-bottom",
      "background--light-tertiary",
      Styles["fake-text"]
    ];

    return classes.join(" ")
  }

  subtitleStyles = () => {
    let classes = [
      "two-thirds",
      "push-bottom",
      "background--lirght-tertiary",
      Styles["fake-text"]
    ];

    return classes.join(" ")
  }

  subsubtitleStyles = () => {
    let classes = [
      "one-fifth",
      "background--light-tertiary",
      Styles["fake-text-small"]
    ];

    return classes.join(" ")
  }

  render () {
    return (
      <div className="card">
        <div className={this.backgroundStyles()}></div>
        <div className="outlined--light soft card__item rounded-bottom">
          <div className={this.titleStyles()}></div>
          <div className={this.subtitleStyles()}></div>
          <div className={this.subsubtitleStyles()}></div>
        </div>
      </div>
    )

  }
}
