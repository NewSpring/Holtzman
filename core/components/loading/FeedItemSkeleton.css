.load-item,
.fake-text {
  background: linear-gradient(270deg, #dddddd, #bcbcbc, #dddddd);
  -webkit-animation: ColorAnimation 1.25s linear infinite;
  animation: ColorAnimation 1.25s linear infinite;
  -ms-animation: ColorAnimation 1.25s linear infinite;
  background-size: 200%;

}

@keyframes ColorAnimation {
    0%{background-position:100%}

    100%{background-position:-100%}
}

@-webkit-keyframes ColorAnimation {
    0%{background-position:100%}

    100%{background-position:-100%}
}

@-ms-keyframes ColorAnimation {
    0%{background-position:100%}

    100%{background-position:-100%}
}

.fake-text {
  height: 15px;
}

.fake-text-small {
  height: 8px;
}
.fake-text, .fake-text-small{
  border-radius: 2px;
}
