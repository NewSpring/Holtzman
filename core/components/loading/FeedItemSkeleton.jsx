import { Component, PropTypes} from "react"


export default class FeedItemSkeleton extends Component {
  render () {
    return (
      <div className="push-left push-half-bottom">
        <div className="rounded-top load-item ratio--square background--light-tertiary"></div>
        <div className="outlined--light soft card__item rounded-bottom">
          <div className="one-whole push-half-bottom background--light-tertiary fake-text" ></div>
          <div className="two-thirds push-bottom background--light-tertiary fake-text" ></div>
          <div className="one-fifth background--light-tertiary fake-text-small"></div>
        </div>
      </div>

    )

  }
}
