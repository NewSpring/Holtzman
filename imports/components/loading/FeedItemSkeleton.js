import { Component } from "react";
import { css } from "aphrodite";
import Styles from "./FeedItemSkeleton-css";

export default class FeedItemSkeleton extends Component {

  backgroundStyles = () => {
    let classes = [
      "rounded-top",
      "ratio--square",
      "background--light-tertiary",
      css(Styles["load-item"])
    ];

    return classes.join(" ");
  }

  titleStyles = () => {
    let classes = [
      "one-whole",
      "push-half-bottom",
      "background--light-tertiary",
      css(Styles["fake-text"])
    ];

    return classes.join(" ");
  }

  subtitleStyles = () => {
    let classes = [
      "two-thirds",
      "push-bottom",
      "background--lirght-tertiary",
      css(Styles["fake-text"])
    ];

    return classes.join(" ");
  }

  subsubtitleStyles = () => {
    let classes = [
      "one-fifth",
      "background--light-tertiary",
      css(Styles["fake-text-small"])
    ];

    return classes.join(" ");
  }

  render () {
    return (
      <div className="card">
        <div className={this.backgroundStyles()} />
        <div className="outlined--light soft card__item rounded-bottom">
          <div className={this.titleStyles()} />
          <div className={this.subtitleStyles()} />
          <div className={this.subsubtitleStyles()} />
        </div>
      </div>
    );

  }
}
