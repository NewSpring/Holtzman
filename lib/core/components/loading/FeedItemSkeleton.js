"use strict";

exports.__esModule = true;
exports["default"] = undefined;

var _classCallCheck2 = require("babel-runtime/helpers/classCallCheck");

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require("babel-runtime/helpers/possibleConstructorReturn");

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require("babel-runtime/helpers/inherits");

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require("react");

var _FeedItemSkeleton = {
  "load-item": "FeedItemSkeleton__load-item___2a44i",
  "fake-text": "FeedItemSkeleton__fake-text___1DR5P",
  "fake-text-small": "FeedItemSkeleton__fake-text-small___1ARkQ",
  "ColorAnimation": "FeedItemSkeleton__ColorAnimation___gr4HE"
};

var _FeedItemSkeleton2 = _interopRequireDefault(_FeedItemSkeleton);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var FeedItemSkeleton = function (_Component) {
  (0, _inherits3["default"])(FeedItemSkeleton, _Component);

  function FeedItemSkeleton() {
    var _temp, _this, _ret;

    (0, _classCallCheck3["default"])(this, FeedItemSkeleton);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3["default"])(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.backgroundStyles = function () {
      var classes = ["rounded-top", "ratio--square", "background--light-tertiary", _FeedItemSkeleton2["default"]["load-item"]];

      return classes.join(" ");
    }, _this.titleStyles = function () {
      var classes = ["one-whole", "push-half-bottom", "background--light-tertiary", _FeedItemSkeleton2["default"]["fake-text"]];

      return classes.join(" ");
    }, _this.subtitleStyles = function () {
      var classes = ["two-thirds", "push-bottom", "background--lirght-tertiary", _FeedItemSkeleton2["default"]["fake-text"]];

      return classes.join(" ");
    }, _this.subsubtitleStyles = function () {
      var classes = ["one-fifth", "background--light-tertiary", _FeedItemSkeleton2["default"]["fake-text-small"]];

      return classes.join(" ");
    }, _temp), (0, _possibleConstructorReturn3["default"])(_this, _ret);
  }

  FeedItemSkeleton.prototype.render = function render() {
    return React.createElement(
      "div",
      { className: "card" },
      React.createElement("div", { className: this.backgroundStyles() }),
      React.createElement(
        "div",
        { className: "outlined--light soft card__item rounded-bottom" },
        React.createElement("div", { className: this.titleStyles() }),
        React.createElement("div", { className: this.subtitleStyles() }),
        React.createElement("div", { className: this.subsubtitleStyles() })
      )
    );
  };

  return FeedItemSkeleton;
}(_react.Component);

exports["default"] = FeedItemSkeleton;
module.exports = exports['default'];