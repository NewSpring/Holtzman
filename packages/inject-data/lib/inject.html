<script type="text/inject-data"><%= data %></script>
