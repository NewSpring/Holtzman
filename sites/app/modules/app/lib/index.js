

// named collections in EE :(
import Collections from "./collections"

export default {
  Collections
}
