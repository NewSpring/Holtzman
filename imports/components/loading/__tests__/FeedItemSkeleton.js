import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { reset, startBuffering } from "aphrodite/lib/inject";
import FeedItemSkeleton from "../FeedItemSkeleton";

const generateComponent = () => (
  <FeedItemSkeleton />
);

beforeEach(() => {
  reset();
  startBuffering();
});

afterEach(() => {
  reset();
});

it("renders", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("backgroundStyles returns styles", () => {
  const wrapper = shallow(generateComponent());
  const result = wrapper.instance().backgroundStyles();
  expect(result).toMatchSnapshot();
});

it("titleStyles returns styles", () => {
  const wrapper = shallow(generateComponent());
  const result = wrapper.instance().titleStyles();
  expect(result).toMatchSnapshot();
});

it("subtitleStyles returns styles", () => {
  const wrapper = shallow(generateComponent());
  const result = wrapper.instance().subtitleStyles();
  expect(result).toMatchSnapshot();
});

it("subsubtitleStyles returns styles", () => {
  const wrapper = shallow(generateComponent());
  const result = wrapper.instance().subsubtitleStyles();
  expect(result).toMatchSnapshot();
});
