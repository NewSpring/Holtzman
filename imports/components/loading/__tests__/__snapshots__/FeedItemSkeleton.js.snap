exports[`test backgroundStyles returns styles 1`] = `"rounded-top ratio--square background--light-tertiary load-item_1vqkxyp"`;

exports[`test renders 1`] = `
<div
  className="card">
  <div
    className="rounded-top ratio--square background--light-tertiary load-item_1vqkxyp" />
  <div
    className="outlined--light soft card__item rounded-bottom">
    <div
      className="one-whole push-half-bottom background--light-tertiary fake-text_jt72u2" />
    <div
      className="two-thirds push-bottom background--lirght-tertiary fake-text_jt72u2" />
    <div
      className="one-fifth background--light-tertiary fake-text-small_nakaq4" />
  </div>
</div>
`;

exports[`test subsubtitleStyles returns styles 1`] = `"one-fifth background--light-tertiary fake-text-small_nakaq4"`;

exports[`test subtitleStyles returns styles 1`] = `"two-thirds push-bottom background--lirght-tertiary fake-text_jt72u2"`;

exports[`test titleStyles returns styles 1`] = `"one-whole push-half-bottom background--light-tertiary fake-text_jt72u2"`;
